### PR TITLE
Move 3rd party headers to a HELICS include directory subfolder

### PR DESCRIPTION
**Pull request by nightlark**
### Description
If merged this pull request will move the helics_includes c++ compatibility headers and CLI11 into a helics subfolder to avoid overwriting a copy installed on the system. The cereal library included with HELICS gets installed into an "optional" subfolder that a user can add to the include directories for their project if they want to use it.

### Proposed changes
- Move helics_includes headers and CLI11 into the helics/external subfolder
- Change the installed cereal headers location to helics/external/optional
- Update includes path used for the c++ compatibility and CLI11 header to make it clear a system copy isn't getting used

### CODE REVIEW

**Comment by phlptp** [comment]
the C++17 headers seem fine to move those. 
One issue with this is that the installed header path needs to be the same as the one in the build directory.    The specific issue is the valueConverter_impl.hpp  If anyone were to use it it needs to be able to correctly find the cereal library in the build and install path.  Since it is the main reason cereal is being installed so applications making use of HELICS C++ can convert their own types easily.    So I want to make sure that header works on install, and I am not sure it does with these changes.

**Comment by phlptp** [comment]
wondering if it might be better to only install cereal by default for Windows systems.  We need cereal in HELICS to ensure stable versions for the internal serialization, but the install is for additional user types which doesn't have the versioning issue, or at least we can reasonably push that on the user.    Then we just need some compile switches inside valueConverter_impl.hpp and it could use a system copy on anything but Visual Studio by default.

**Comment by nightlark** [comment]
The idea behind including a copy of cereal in the optional folder is that if someone wants to use ValueConverter_impl for serializing a custom data type, then they can add the `helics/external/optional` folder to their include path and they'd have the same version that HELICS was built with.

It seems like cereal expects the folder it is contained in to be an include directory (`#include external/cereal/detail/blah.hpp` gives an error because cereal headers include each other using `cereal/blah.hpp`), so I don't think a compile switch is even needed -- on Linux they just need to install a copy of cereal in their include path if they want to use ValueConverter_impl; I wasn't sure if it made sense to have cereal in the include directories for the HELICS target by default, in case they already had a system copy (maybe exporting a target just for cereal + HELICS_CEREAL_INCLUDE_DIR variable?).

An easy option to convert custom types makes sense, I'm not sure cereal should be encouraged as the way to do serialization since it requires at least C++11 (integrating federates that use other language bindings later would be a challenge).

**Comment by phlptp** [comment]
As far as cereal I was considering using it but mainly for the direct apps and other applications that can be tightly integrated into the C++ code.    HELICS-BBX, GridDyn, HELICS-FMI,  and a few others, so wasn't sure how many others would use it.  

That said, I kind of like the idea of an optional install for cereal that is off by default except for MSVC builds and a separate target that gets generated with the appropriate include file for cereal if the CMAKE option is enabled.

**Comment by nightlark** [comment]
I think a good guideline is if someone might want to access to the value from a custom federate (e.g. control message, voltage value), then a more portable format should be used. If the federate on the other end is always going to be the same (e.g. another instance of the same program) then cereal is okay.

I added an exported `HELICS::cereal` CMake target that adds the `helics/external/optional` include directory, and a component in cpack to disable installing cereal -- is disabling the component by default something we want to do, or is  `target_link_libraries(<target> HELICS::cereal)` enough of an opt-in?

**Comment by phlptp** [comment]
@nightlark I made a PR onto yours about adding an option around it as well so it can be completely turned off.

File: CMakeLists.txt [hunk 1]
```diff
@@ -809,7 +819,7 @@ if(ENABLE_PACKAGE_BUILD)
         "Headers for linking and compiling with HELICS"
     )
     set(CPACK_COMPONENT_RUNTIME_DESCRIPTION "Runtime libraries for HELICS")
-
+    set(CPACK_COMPONENT_CEREAL_DESCRIPTION "cereal C++11 serialization library [no support for other language interfaces]")
```
[the remark this answers is not in the document]
Comment (nightlark) [reply CMakeLists.txt]: It's setting a variable, that I guess never gets used; I'll check if there is a warning about the variable not being used.

[the remark this answers is not in the document]
Comment (nightlark) [reply CMakeLists.txt]: It looks like it is okay, no warnings are printed and it doesn't show up in the package list for installers unless the cereal install targets exist.

**Pull request merged**